src/psyclone/transformations.py

```diff
@@ -340,67 +379,124 @@ def name(self):
         ''' Returns the name of this transformation as a string.'''
         return "DynamoLoopFuse"
 
-    def apply(self, node1, node2, same_space=False):
-        '''
-        Fuse the two Dynamo loops represented by :py:obj:`node1` and
-        :py:obj:`node2`. The optional same_space flag asserts that an
-        unknown iteration space (i.e. any_space) matches the other
-        iteration space. This is set at the users own risk.
+    def _validate(self, node1, node2):
+        ''' Performs various checks to ensure that it is valid to apply
+        the DynamoLoopFuseTrans transformation to the supplied Loops.
 
-        :param node1: First Loop to fuse.
+        :param node1: the first Loop to fuse.
         :type node1: :py:class:`psyclone.dynamo0p3.DynLoop`
-        :param node2: Second Loop to fuse.
+        :param node2: the second Loop to fuse.
         :type node2: :py:class:`psyclone.dynamo0p3.DynLoop`
-        :returns: two-tuple of modified Schedule and Memento
-        :rtype: :py:class:`psyclone.psyGen.Schedule`, \
-                :py:class:`psyclone.undoredo.Memento`
-        :raises TransformationError: if either of the supplied loops contains \
+        :returns: two-tuple of the modified Schedule and a record of \
+                  the transformation.
+        :rtype: (:py:class:`psyclone.psyGen.Schedule`, \
+                 :py:class:`psyclone.undoredo.Memento`)
+
+        :raises TransformationError: if either of the supplied Loops contains \
                                      an inter-grid kernel.
+        :raises TransformationError: if one or both function spaces have \
+                                     invalid names.
+        :raises TransformationError: if the `same_space`flag was set, but \
+                                     does not apply because neither field \
+                                     is on `ANY_SPACE`.
+        :raises TransformationError: if one or more of the iteration spaces
+                                     is unknown (`ANY_SPACE`) and the \
+                                     `same_space` optional argument is not \
+                                     set to `True`.
+        :raises TransformationError: if the Loops are over different spaces \
```

Comment:
As discussed today at the Telco, this is fine as we are looping over cells in the user-supplied kernels. If we were looping over dofs than two different discontinuous spaces would have different number of dofs so it would not be safe to fuse the loops. However, looping over dofs is currenlty enabled only in built-ins which are on `ANY_SPACE` anyway.